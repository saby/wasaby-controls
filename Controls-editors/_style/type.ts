
import {
    ObjectType,
} from 'Types/meta';
import StyleEditor from './editor';

/**
 * Метатип стилей
 */
export const StyleType = ObjectType.editor(() => Promise.resolve(StyleEditor));
StyleType.getEditor().load();
