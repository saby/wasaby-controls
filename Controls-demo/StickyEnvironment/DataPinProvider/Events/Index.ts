import { Memory } from 'Types/source';
import { SyntheticEvent } from 'UI/Vdom';
import { IEdgesData } from 'Controls/stickyEnvironment';
import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import { generateData } from 'Controls-demo/list_new/DemoHelpers/DataCatalog';
import * as template from 'wml!Controls-demo/StickyEnvironment/DataPinProvider/Events/Index';

export default class Index extends Control<IControlOptions> {
    protected _template: TemplateFunction = template;

    protected _topEdgeData: number;
    protected _bottomEdgeData: number;

    protected _viewSource: Memory;
    protected _navigation: object = {
        source: 'page',
        view: 'infinity',
        sourceConfig: {
            page: 10,
            pageSize: 50,
            hasMore: false,
        },
    };

    private _dataArray: { key: number; title: string }[] = generateData<{
        key: number;
        title: string;
    }>({
        count: 1000,
        entityTemplate: { title: 'lorem' },
    });

    protected _beforeMount(): void {
        this._viewSource = new Memory({
            keyProperty: 'key',
            data: this._dataArray,
        });
    }

    protected _onEdgesDataChanged(
        event: SyntheticEvent,
        data: IEdgesData<number>
    ): void {
        this._topEdgeData = data.top.above || data.top.below;
        this._bottomEdgeData = data.bottom.above;
    }
}
