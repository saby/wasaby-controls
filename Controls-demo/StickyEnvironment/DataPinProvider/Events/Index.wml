<div class="controls-demo_Wrapper controlsDemo_fixedWidth500">
   <div data-qa="controls-demo__text_top">Top edge data id: {{_topEdgeData}}</div>
   <div data-qa="controls-demo__text_bottom">Bottom edge data id: {{_bottomEdgeData}}</div>

   <Controls.stickyEnvironment:DataPinProvider on:edgesDataChanged="_onEdgesDataChanged()">
      <Controls.scroll:Container attr:style="height: 500px;">
         <Controls.list:View
            source="{{_viewSource}}"
            keyProperty="key"
            navigation="{{_navigation}}"
         >
            <ws:virtualScrollConfig page="{{50}}" pageSize="{{50}}"/>
            <ws:itemTemplate>
               <ws:partial template="Controls/list:ItemTemplate">
                  <ws:contentTemplate>
                     <Controls.stickyEnvironment:DataPinContainer data="{{itemTemplate.item.contents.get('key')}}">
                        <strong>Запись с id="{{itemTemplate.item.contents.get('key')}}"</strong>
                     </Controls.stickyEnvironment:DataPinContainer>

                     {{itemTemplate.item.contents.get('title')}}
                     <i data-qa="Controls-Demo_StickyEnvironment__hooks">h</i>
                  </ws:contentTemplate>
               </ws:partial>
            </ws:itemTemplate>
         </Controls.list:View>
      </Controls.scroll:Container>
   </Controls.stickyEnvironment:DataPinProvider>
</div>
