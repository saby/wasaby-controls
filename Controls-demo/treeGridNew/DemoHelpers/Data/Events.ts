import { IColumn } from 'Controls/grid';
import { IData } from '../Interface';

export const Events = {
    getColumns: (): IColumn[] => {
        return [
            {
                displayProperty: 'title',
                width: '',
            },
        ];
    },
    getDataBeforeExpanded: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед разворотом узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataBeforeCollapsed: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Асинхронная обработка перед сворачиванием узла',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataCollapsedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно свернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
    getDataExpandedItemsChanged: (): IData[] => {
        return [
            {
                key: 1,
                title: 'Этот узел нельзя развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 11, title: 'Лист 1-1', parent: 1, nodeType: null },
            { key: 12, title: 'Лист 1-2', parent: 1, nodeType: null },
            {
                key: 2,
                title: 'Этот узел можно развернуть',
                parent: null,
                nodeType: true,
            },
            { key: 21, title: 'Лист 2-1', parent: 2, nodeType: null },
            { key: 22, title: 'Лист 2-2', parent: 2, nodeType: null },
        ];
    },
};
