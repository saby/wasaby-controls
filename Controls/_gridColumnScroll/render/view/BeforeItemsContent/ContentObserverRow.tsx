import { ContentObserverComponent } from 'Controls/columnScrollReact';
import useGridSeparatedRowStyles, {
    IUseGridSeparatedRowStylesProps,
} from './useGridSeparatedRowStyles';

// TODO: Убрать после
//  https://online.sbis.ru/opendoc.html?guid=ad3356a9-0caf-4acf-8679-3dd6b6a3d711&client=3
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IContentObserverRowProps extends Omit<IUseGridSeparatedRowStylesProps, 'part'> {
}

export default function ContentObserverRow(
    props: IContentObserverRowProps
): JSX.Element {
    const { fixedCellStyle, scrollableCellStyle } = useGridSeparatedRowStyles(props);

    return (
        <div className="tw-contents">
            <ContentObserverComponent
                fixedDiv={<div style={fixedCellStyle} />}
                scrollableDiv={<div style={scrollableCellStyle} />}
            />
        </div>
    );
}
