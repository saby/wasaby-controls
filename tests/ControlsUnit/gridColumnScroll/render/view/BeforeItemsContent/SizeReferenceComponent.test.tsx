/**
 * @jest-environment jsdom
 */
import type { IContentObserverComponentProps } from 'Controls/_columnScrollReact/SizeObservers/ContentObserver';

jest.mock('Controls/_columnScrollReact/SizeObservers/ContentObserver', () => {
    return {
        __esModule: true,
        default: (props: IContentObserverComponentProps) => {
            return (
                <>
                    {props.startFixedDiv}
                    {props.scrollableDiv}
                    {props.endFixedDiv}
                </>
            );
        },
    };
});
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';

import ContentObserverRow from 'Controls/_gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow';

describe('ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn.', () => {
        const stickyColumnsCountVariants = [0, 1, 3];
        const hasMultiSelectColumnVariants = [true, false];
        const hasResizerVariants = [true, false];

        const test = (
            stickyColumnsCount: number,
            hasMultiSelectColumn: boolean,
            hasResizer: boolean
        ) => {
            it(
                `stickyColumnsCount=${stickyColumnsCount}, ` +
                    `hasMultiSelectColumn=${hasMultiSelectColumn}, ` +
                    `hasResizer=${hasResizer},`,
                () => {
                    const { asFragment } = render(
                        <ContentObserverRow
                            hasResizer={hasResizer}
                            stickyColumnsCount={stickyColumnsCount}
                            endStickyColumnsCount={0}
                            hasMultiSelectColumn={hasMultiSelectColumn}
                        />,
                        {
                            container,
                        }
                    );

                    expect(asFragment()).toMatchSnapshot();
                }
            );
        };

        stickyColumnsCountVariants.forEach((stickyColumnsCount) => {
            hasMultiSelectColumnVariants.forEach((hasMultiSelectColumn) => {
                hasResizerVariants.forEach((hasResizer) => {
                    test(stickyColumnsCount, hasMultiSelectColumn, hasResizer);
                });
            });
        });
    });
});
