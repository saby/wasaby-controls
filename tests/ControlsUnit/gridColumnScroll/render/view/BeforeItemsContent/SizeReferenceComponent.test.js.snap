// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=0, hasMultiSelectColumn=false, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 1;"
    />
    <div
      style="grid-column-start: 1; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=0, hasMultiSelectColumn=false, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 1;"
    />
    <div
      style="grid-column-start: 2; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=0, hasMultiSelectColumn=true, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 2;"
    />
    <div
      style="grid-column-start: 2; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=0, hasMultiSelectColumn=true, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 2;"
    />
    <div
      style="grid-column-start: 3; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=1, hasMultiSelectColumn=false, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 2;"
    />
    <div
      style="grid-column-start: 2; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=1, hasMultiSelectColumn=false, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 2;"
    />
    <div
      style="grid-column-start: 3; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=1, hasMultiSelectColumn=true, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 3;"
    />
    <div
      style="grid-column-start: 3; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=1, hasMultiSelectColumn=true, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 3;"
    />
    <div
      style="grid-column-start: 4; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=3, hasMultiSelectColumn=false, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 4;"
    />
    <div
      style="grid-column-start: 4; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=3, hasMultiSelectColumn=false, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 4;"
    />
    <div
      style="grid-column-start: 5; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=3, hasMultiSelectColumn=true, hasResizer=false, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 5;"
    />
    <div
      style="grid-column-start: 5; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;

exports[`ControlsUnit/gridColumnScroll/render/view/BeforeItemsContent/ContentObserverRow Рендер при различных значениях опций stickyColumnsCount и hasMultiSelectColumn. stickyColumnsCount=3, hasMultiSelectColumn=true, hasResizer=true, 1`] = `
<DocumentFragment>
  <div
    class="tw-contents"
  >
    <div
      style="grid-column-start: 1; grid-column-end: 5;"
    />
    <div
      style="grid-column-start: 6; grid-column-end: -1;"
    />
  </div>
</DocumentFragment>
`;
